feat(segmentation): handle pre-authenticated users

adekbadek opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes #903.

### How to test the changes in this Pull Request:

1. Create a prompt segmented to users who have an account
2. Enable Reader Activation (`NEWSPACK_EXPERIMENTAL_READER_ACTIVATION` env. variable)
3. In a new session, observe the prompt is not displayed
4. Pre-authenticate by entering existing user's email in Registration block
5. Navigate to another page, observe the prompt displayed

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo reviewed (approved):
Works as described!

Is there any way to process the registration such that the reader is segmented properly on the very next pageview after registering?

adekbadek commented:
> Is there any way to process the registration such that the reader is segmented properly on the very next pageview after registering?

@dkoo – actually, this can be handled via a WP hook! I've reverted the original commit and took this approach. I've updated the PR description to reflect that.
